traitsui/testing/tester/wx/default_registry.py

```diff
@@ -19,7 +26,48 @@ def get_default_registry():
     -------
     registry : TargetRegistry
         The default registry containing implementations for TraitsUI editors
-        that is wx specific.  
+        that is wx specific.
+    """
+    registry = get_wx_object_registry()
+
+    button_editor.register(registry)
+
+    return registry
+
+
+def get_wx_object_registry():
+    """ Creates a generic registry for handling/solving wx objects. (i.e.
+    this registry is independent of TraitsUI)
+
+    Returns
+    -------
+    registry : TargetRegistry
+        Registry containing wx specific generic handlers and solvers.
     """
     registry = TargetRegistry()
+
+    registry.register_handler(
+        target_class=wx.Button,
+        interaction_class=command.MouseClick,
+        handler=helpers.mouse_click_button
+    )
+
+    registry.register_handler(
+        target_class=wx.Button,
+        interaction_class=query.DisplayedText,
+        handler=lambda wrapper, _: wrapper.target.GetLabel()
+    )
+
+    registry.register_handler(
+        target_class=wx.Window,
+        interaction_class=command.MouseClick,
+        handler=helpers.mouse_click_ImageButton
+    )
+
+    registry.register_handler(
+        target_class=wx.Window,
+        interaction_class=query.DisplayedText,
+        handler=lambda wrapper, _: wrapper.target.GetLabel()
+    )
```

Comment:
`wx.Window` [is extremely broad in its functionality](https://wxpython.org/Phoenix/docs/html/wx.Window.html), so I am not sure it belongs in this registry for reuse purposes, despite it being a wx object.

We discussed the cost and benefit of the approach of DefaultTarget + generic registry:
The cost is a more obscure code path that is difficult to do static code analysis on. The benefit is the ability to reuse interactions already defined on common and simple widgets.

The widgets in this registry should be so simple that it cannot be broken down into smaller pieces, there should be only one place else a mouse click or key click could occur, and expected outcome of these actions should be unambiguous. Simple things like button and textbox are good candidates for that, but a container that can be used for many different use cases is not. If we are good with deciding what interactions should be elevated to be shared status, this would be okay, but if we are not, tracking these mistakes will be hard.

That's the risk factor I feared: It is so tempting to add more complicated components to this registry where they don't belong. I think we prove that this trap is real.

The elaborate comments are initial indications to the cost, this `wx.Window` is the last straw. The combined cost and risk make me think that this DefaultTarget and the `get_wx_object_registry` / `get_qobject_registry` is not worth the benefit of reuse, so let's revert this. Instead of using a DefaultTarget for delegation, we should register the MouseClick and DisplayedText handlers to the (simple/custom) ButtonEditor directly. This will keep the logic and pattern simple to follow - developers will not need to think very hard about this.

It is really good that this happens in such an early stage, thank you very much!  Now we can make this conscious choice with evidence of what could go wrong had we kept this pattern.

Most of the tests in this PR can be kept. The new tests we added for the nested-try-block can go. The tests in `test_default_registry` will probably need to be modified to call the functions from `helpers` instead (and maybe moved to `test_helpers`).

Reply:
I realize that for the current use case, I believe these handlers can easily be modified to use ImageButton as the target class instead of wx.Window.  After I do that I will move them into the Button Editor implementation rather than the get_wx_object_registry.

Reply:
I can still undo the Default Target delegation as well.

Reply:
Thank you. Let's undo the Default Target delegation if that's okay for you.